compiler/damlc/tests/daml-test-files/InterfaceDoubleSpend.daml

```diff
@@ -1,13 +1,19 @@
 -- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
--- TODO https://github.com/digital-asset/daml/issues/12697
+{-# LANGUAGE ApplicativeDo #-}
```

Comment:
Update is an Action, so it supports the full normal `do` syntax. Commands however are not, they cannot support interdependence nor the usual Action flow. They are however an Applicative, 1 step below Action. Applicative doesn't _usually_ support a do syntax, but with the ApplicativeDo extension, and by following a few rules, we can use it as such

Reply:
So in this case, line 61-64 would break without this extension, as we're doing multiple actions within a Commands do block.
You can also notice the `pure ()` there. With an Action, we could simply use `void` on the line above, but ApplicativeDo would not recognise this

Reply:
Wait, unless you mean the test itself? Since it was your comment below that says to remove it?
If we have solid proof that this behaviour is tested elsewhere, it can maybe go, it does feel bad removing tests though

Reply:
I was speaking about the `ApplicativeDo`. Thanks for the explication.